### PR TITLE
kafka: update sarama dependence

### LINKED ISSUE #20: shopify/samara has been replaced by IBM/sarama, please update kafka plugin's dependency

### PR DESCRIPTION
**Pull request by Leoyoungxh**
Update github.com/Shopify/sarama v1.29.1 to github.com/IBM/sarama v1.40.1

Fixes #20

### CODE REVIEW

File: kafka/client.go [hunk 1]
```diff
@@ -4,7 +4,7 @@ import (
 	"context"
 	"fmt"
 
-	"github.com/Shopify/sarama"
+	"github.com/IBM/sarama"
```
Comment (liuzengh) [inline kafka/client.go:7]: Has there been a change in API compatibility? eg: `sarama.RecordHeader`
```golang
// Client kafka interface
type Client interface {
	Produce(ctx context.Context, key, value []byte,
		headers ...sarama.RecordHeader) error
	SendMessage(ctx context.Context, topic string, key, value []byte,
		headers ...sarama.RecordHeader) (partition int32, offset int64, err error)
	AsyncSendMessage(ctx context.Context, topic string, key, value []byte,
		headers ...sarama.RecordHeader) (err error)
	// SendSaramaMessage produce sarama native messages directly
	SendSaramaMessage(ctx context.Context, sMsg sarama.ProducerMessage) (partition int32, offset int64, err error)
```

Comment (Leoyoungxh) [reply kafka/client.go:7]: The user needs to replace "github.com/Shopify/sarama" with "github.com/IBM/sarama". 

The "github.com/Shopify/sarama" is no longer in use and we will specify it in the CHANGELOG.

**Review by liuzengh — APPROVED** [review]
LGTM

**Comment by liuzengh** [comment]
Merged, and a new tag kafka/1.1.0 has been released.&nbsp;https://github.com/trpc-ecosystem/go-database/releases/tag/kafka%2F1.1.0&nbsp;


good
***@***.***







&nbsp;




------------------&nbsp;原始邮件&nbsp;------------------
发件人: ***@***.***&gt;; 
发送时间: 2023年12月21日(星期四) 晚上9:04
收件人: ***@***.***&gt;; 
抄送: ***@***.***&gt;; "Review ***@***.***&gt;; 
主题: Re: [trpc-ecosystem/go-database] kafka: update sarama dependence (PR #21)





 
is there any progress ?
 
&mdash;

**Pull request merged**